### ISSUE TITLE
Detect incorrect chaining of `if` and `if let` conditions and recover gracefully

### ISSUE THREAD

**Issue by estebank** [rust-lang/rust#103381]
Given

```rust
fn foo(b: bool, x: Option<u32>) {
    if b && if let Some(x) = x {}
}
fn main() {
    if true && if true { true }
}
```

[the current output is](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=28db46bf3ce2d2c7d3dae7305740e8ee)

```
error: expected `{`, found `}`
 --> src/main.rs:3:1
  |
3 | }
  | ^ expected `{`
  |
note: the `if` expression is missing a block after this condition
 --> src/main.rs:2:8
  |
2 |     if b && if let Some(x) = x {}
  |        ^^^^^^^^^^^^^^^^^^^^^^^^^^

error: expected `{`, found `}`
 --> src/main.rs:6:1
  |
6 | }
  | ^ expected `{`
  |
note: the `if` expression is missing a block after this condition
 --> src/main.rs:5:8
  |
5 |     if true && if true { true }
  |        ^^^^^^^^^^^^^^^^^^^^^^^^
```

We should detect that what was intended was

```rust
fn foo(b: bool, x: Option<u32>) {
    if b && let Some(x) = x {}
}
fn main() {
    if true && true { true }
}
```

_https://github.com/rust-lang/rust/issues/53667#issuecomment-1228604924_
